packages/sanity/src/_internal/cli/actions/schema/storeSchemasAction.ts

```diff
@@ -0,0 +1,165 @@
+import {type CliCommandContext, type CliOutputter} from '@sanity/cli'
+import {type SanityClient} from '@sanity/client'
+import chalk from 'chalk'
+import partition from 'lodash/partition'
+
+import {
+  type ManifestWorkspaceFile,
+  SANITY_WORKSPACE_SCHEMA_TYPE,
+  type StoredWorkspaceSchema,
+} from '../../../manifest/manifestTypes'
+import {type SchemaStoreContext} from './schemaStoreTypes'
+import {createManifestExtractor, isManifestExtractSatisfied} from './utils/mainfestExtractor'
+import {type CreateManifestReader, createManifestReader} from './utils/manifestReader'
+import {createSchemaApiClient} from './utils/schemaApiClient'
+import {
+  FlagValidationError,
+  parseStoreSchemasConfig,
+  type StoreSchemaCommonFlags,
+  throwWriteProjectIdMismatch,
+} from './utils/schemaStoreValidation'
+import {getWorkspaceSchemaId} from './utils/workspaceSchemaId'
+
+export interface StoreSchemasFlags extends StoreSchemaCommonFlags {
+  'workspace'?: string
+  'id-prefix'?: string
+  'schema-required'?: boolean
+}
+
+export default function storeSchemasActionForCommand(
+  flags: StoreSchemasFlags,
+  context: CliCommandContext,
+): Promise<'success' | 'failure'> {
```

Comment:
this seems a bit unconventional to me – why not `Promise<void>` and take the error path on failure?

Reply:
It is because of the `--schema-required` flag; when it is true, the action is throwing, when it is false, the action will never throw. 
Would you allow it with a comment? I could encapsulate as a result type, and add a tsdoc with a reason to the type?

Reply:
I'll push the union type with tscomment, and you can let me know what you think

Reply:
That makes sense, thanks for adding a comment!